ocaml/xapi/xapi_ha_vm_failover.ml

```diff
@@ -175,60 +193,452 @@ let order_f (_, vm_rec) =
 
 let ( $ ) x y = x y
 
-(*****************************************************************************************************)
-(* Planning code follows                                                                             *)
-
 (* Compute the total memory required of a VM (Running or not) *)
 let total_memory_of_vm ~__context policy snapshot =
   let main, shadow =
     Memory_check.vm_compute_start_memory ~__context ~policy snapshot
   in
   Int64.add main shadow
 
-(** Return a VM -> Host plan for the Host.evacuate code. We assume the VMs are all agile. The returned plan may
-    be incomplete if there was not enough memory. *)
-let compute_evacuation_plan ~__context total_hosts remaining_hosts
-    vms_and_snapshots =
-  let hosts =
-    List.map
-      (fun host ->
-        ( host
-        , Memory_check.host_compute_free_memory_with_maximum_compression
-            ~__context ~host None
+(* Used to sort pairs into increasing order of their second component *)
+let bigger_than (_, a) (_, b) = compare a b
+
+let fst_host_can_run vm_can_boot_on_host vm available_hosts_increasing =
```

Comment:
This one line function is no necessary.

Reply:
It is used in 2 places, and I think with this function name, it might make the code easier to understand.